src/main/java/sirius/web/util/BarcodeController.java

```diff
@@ -59,39 +83,94 @@ private void barcode(WebContext webContext, BarcodeFormat format) throws WriterE
         int height = webContext.getFirstFilled("h", "height").asInt(200);
         String content = webContext.getFirstFilled("c", "content").asString();
         if (Strings.isEmpty(content)) {
-            webContext.respondWith().direct(HttpResponseStatus.BAD_REQUEST, "Usage: /barcode?type=qr&content=...&w=200&h=200");
+            webContext.respondWith()
+                      .direct(HttpResponseStatus.BAD_REQUEST, "Usage: /barcode?type=qr&content=...&w=200&h=200");
             return;
         }
 
+        content = alignContentForItfFormat(content, webContext.get("type").asString());
+
+        format = useItfFormatForGtin14(content, format);
+
         String fileType = webContext.getFirstFilled("fileType").asString("jpg");
         Writer writer = determineWriter(format);
         BitMatrix matrix = writer.encode(content, format, width, height);
         try (OutputStream out = webContext.respondWith()
-                                   .infinitelyCached()
-                                   .outputStream(HttpResponseStatus.OK,
-                                                 MimeHelper.guessMimeType("barcode." + fileType))) {
+                                          .infinitelyCached()
+                                          .outputStream(HttpResponseStatus.OK,
+                                                        MimeHelper.guessMimeType("barcode." + fileType))) {
             MatrixToImageWriter.writeToStream(matrix, fileType, out);
         }
     }
 
-    private BarcodeFormat determineFormat(String format) {
-        return switch (format) {
-            case "qr" -> BarcodeFormat.QR_CODE;
-            case "code128" -> BarcodeFormat.CODE_128;
-            case "ean" -> BarcodeFormat.EAN_13;
-            case "datamatrix" -> BarcodeFormat.DATA_MATRIX;
+    /**
+     * Generates an image of a barcode
+     *
+     * @param type    the desired barcode type
+     * @param content the content of the barcode
+     * @return a barcode of the given data as a 200x200 px image
+     * @throws WriterException if generating the image fails
+     */
+    public static Image generateBarcodeImage(String type, String content) throws WriterException {
+        if (!NUMERIC.matcher(content).matches()) {
+            // contains characters other than digits 0-9 -> directly return a blank image to prevent running into exception
+            return new BufferedImage(200, 200, BufferedImage.TYPE_BYTE_GRAY);
+        }
+
+        BarcodeFormat format = determineFormat(type);
+
+        content = alignContentForItfFormat(content, type);
+
+        format = useItfFormatForGtin14(content, format);
+
+        Writer writer = determineWriter(format);
+        BitMatrix matrix = writer.encode(content, format, 200, 200);
```

Comment:
The cssWidth and cssHeight needs to be used here. At least that was the case in my tests